.github/workflows/build-test-publish.yml

```diff
@@ -0,0 +1,59 @@
+name: "Build Test Publish"
+
+on:
+  workflow_dispatch:
+    # Just in case
+
+  pull_request:
+    branches: [main]
+
+  push:
+    branches: [main]
+    tags:
+      - 'v*'
+      - "!**-test"
+      - "!**-dev"
+
+
+jobs:
+  build-modules:
+    uses: infinity-swap/.github/.github/workflows/build-n-test.yml@v0.1
+    with:
+      test-script: ./scripts/cargo.test.sh
+      skip-test: ${{ github.ref_type == 'tag' }}
+      output-artifact: is20-modules
+    secrets:
+      gh_token: ${{ secrets.GH_PKG_TOKEN }}
+      gh_login: ${{ secrets.GH_PKG_LOGIN }}
+  
+  publish-prepare:
```

Comment:
I think the logic for publich-prepare can be moved to the .github publishing workflow as the checks such as 
 ${{ github.ref_type == 'branch' && github.ref == 'refs/heads/main' }} will be possible.

Reply:
Probably but I have no idea how to pass output variables from reusable forflow to the next job

Reply:
Found how to do it.
Yes it is possible to hide some logic but still we should pass all input arguments there